scripts/vscripts/tf2ware_ultimate/minigames/pick_up_that_can.nut

```diff
@@ -0,0 +1,222 @@
+		solid = 6
+    })
+
+    if (Outline)
+    {
+        local glow = Ware_CreateEntity("tf_glow")
+        glow.KeyValueFromString("GlowColor", "0 255 0 255");
+        SetPropEntity(glow, "m_hTarget", trashcan);
+    }
+}
+
+function SpawnCans()
+{
+	local mins = Ware_MinigameLocation.mins
+	local maxs = Ware_MinigameLocation.maxs
+
+	foreach (player in Ware_MinigamePlayers)
+	{
+        local origin = Vector(
+		RandomFloat(Ware_MinigameLocation.mins.x + CanOffset, Ware_MinigameLocation.maxs.x - CanOffset),
+		RandomFloat(Ware_MinigameLocation.mins.y + CanOffset, Ware_MinigameLocation.maxs.y - CanOffset),
+		Ware_MinigameLocation.center.z + CanHeightOffset)
+        
+		local can = Ware_SpawnEntity("prop_physics",
+		{
+			model = can_model
+			origin = origin
+		})
+        can.SetCollisionGroup(COLLISION_GROUP_DEBRIS_TRIGGER)
+        can.ValidateScriptScope()
+        can.GetScriptScope().LastHolder <- null
+
+        if (Outline)
+        {
+            local glow = Ware_CreateEntity("tf_glow")
+            glow.KeyValueFromString("GlowColor", "255 0 111 255");
+            SetPropEntity(glow, "m_hTarget", can);
+        }
+	}
+}
+
+function SpawnTriggerMultiple()
+{
+    local center = Ware_MinigameLocation.center
+    local trigger_multiple = Ware_SpawnEntity("trigger_multiple",
+    {
+        targetname = "trashcan_trigger"
+        origin     = center
+        spawnflags = 8
+    })
+    trigger_multiple.SetSize(Vector(-5, -5, 0), Vector(5, 5, 24))
+    trigger_multiple.SetSolid(2)
+
+    trigger_multiple.ValidateScriptScope()
+	trigger_multiple.GetScriptScope().OnStartTouch <- OnTriggerTouch
+	trigger_multiple.ConnectOutput("OnStartTouch", "OnStartTouch")
+}
+
+function SpawnTriggerPush()
+{
+    local center = Ware_MinigameLocation.center
+    local trigger_push = Ware_SpawnEntity("trigger_push", 
+    {
+        targetname = "trashcan_push"
+        origin     = center
+        pushdir    = QAngle(0, RandomFloat(0, 360), 0)
+        speed      = 1000
+        spawnflags = 1
+    })
+    trigger_push.SetSize(Vector(-20, -20, 0), Vector(20, 20, 50))
+    trigger_push.SetSolid(2)
+    EntFireByHandle(trigger_push, "Disable", "", 0.5, null, null)
+}
+
+function OnTriggerTouch()
+{
+    activator.ValidateScriptScope()
+    local player = activator.GetScriptScope().LastHolder
+    if (player)
+        Ware_PassPlayer(player, true)
+    
+    activator.Destroy() //If the cans are not removed, they will lag the server due to physics calculations..
+}
+
+// Inspired by Prop Kill (https://github.com/Batfoxkid/TF2-Prop-Kill)
+function PlayerThink()
+{
+    local alive = GetPropInt(self, "m_lifeState") == 0
+    local buttons = alive ? GetPropInt(self, "m_nButtons") : 0;
+    local newButtons = buttons & ~LastButtons;
+
+    local MOUSE1 = (newButtons & Constants.FButtons.IN_ATTACK);
+    local MOUSE2 = (newButtons & Constants.FButtons.IN_ATTACK2);
+
+    if (MOUSE1 || MOUSE2)
+    {
+        if (PickedProp != self && PickedProp.IsValid())

```

Comment:
Just a note that you will need to use something like `local minidata = Ware_GetPlayerMiniData(self)` and then access these as `minidata.PickedProp` instead